vault/resource_plugin_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+const envPluginCommand = "VAULT_PLUGIN_COMMAND"
+
+func TestPlugin(t *testing.T) {
+	const (
+		typ     = "auth"
+		version = "v1.0.0"
+		args    = `["--foo"]`
+		env     = `["FOO=BAR"]`
+
+		argsUpdated = `["--bar"]`
+		envUpdated  = `["FOO=BAZ"]`
+	)
+
+	destName := acctest.RandomWithPrefix("tf/plugin")
+
+	resourceName := "vault_plugin.test"
+	sha256 := strings.Repeat("01234567", 8)
+	sha256Updated := strings.Repeat("12345678", 8)
+	cmd := os.Getenv(envPluginCommand)
```

Comment:
nit: can we add in a comment what this parameter should look like? Can we pass in any random value or does it have to be something specific and set up as a prerequisite for the test? Thanks!

Reply:
That's right, added a comment 👍